applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_broadcast.c

```diff
@@ -42,6 +42,33 @@ struct broadcast_source {
 	uint32_t broadcast_id;
 };
 
+static void scan_restart_worker(struct k_work *work)
+{
+	int ret;
+
+	/* Delete pending PA sync before restarting scan */
+	ret = bt_mgmt_pa_sync_delete(pa_sync);
+	if (ret) {
+		LOG_WRN("Failed to delete pending PA sync: %d", ret);
+	}
+
+	ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_BROADCAST, NULL);
+	if (ret) {
+		LOG_WRN("Failed to restart scanning for braodcast: %d", ret);
+	}
+}
+
+K_WORK_DEFINE(scan_restart_work, scan_restart_worker);
+
+static void pa_sync_timeout(struct k_timer *timer)
+{
+	LOG_WRN("PA sync create timed out, restarting scanning");
+
+	k_work_submit(&scan_restart_work);
```

Comment:
I think using delayable work instead of timer + work is cleaner.
k_work_schedule() and k_work_cancel_delayable()

Reply:
See e.g. unicast_client.c for reference

Reply:
Not sure I agree. It requires a separate init function to be run and the timer has the benefit of being able to be started multiple times and it will just reset the timer. If you want similar behaviour for k_work_delayable I believe you have to first cancel k_work if it is pending and then reschedule

Reply:
Using a timer is also closer to a traditional watchdog, so I think it is easier to read